documentation/api/v1/module_api_doc.rb

```diff
@@ -0,0 +1,341 @@
+      'Dylan Davis <dylan.davis@risksense.com>',
+      'Equation Group',
+      'Shadow Brokers',
+      'thelightcosine'
+  ]
+  REFERENCES_EXAMPLE = [
+      'AKA-ETERNALBLUE',
+      'MSB-MS17-010',
+      'CVE-2017-0143',
+      'CVE-2017-0144',
+      'CVE-2017-0145',
+      'CVE-2017-0146',
+      'CVE-2017-0147',
+      'CVE-2017-0148',
+      'URL-https://github.com/RiskSense-Ops/MS17-010'
+  ]
+  IS_SERVER_EXAMPLE = true
+  IS_CLIENT_EXAMPLE = false
+  PLATFORM_EXAMPLE = 'Windows'
+  ARCH_EXAMPLE = ''
+  RPORT_EXAMPLE = '445'
+  MOD_TIME_EXAMPLE = '2018-07-10T17:39:42.000-05:00'
+  REF_NAME_EXAMPLE = 'windows/smb/ms17_010_eternalblue'
+  PATH_EXAMPLE = '/modules/exploits/windows/smb/ms17_010_eternalblue.rb'
+  IS_INSTALL_PATH_EXAMPLE = true
+  TARGETS_EXAMPLE = [
+      'Windows 7 and Server 2008 R2 (x64) All Service Packs'
+  ]
+
+
+  # Swagger documentation for Module model
+  swagger_schema :Module do
+    property :name, type: :string, description: NAME_DESC, example: NAME_EXAMPLE
+    property :full_name, type: :string, description: FULL_NAME_DESC, example: FULL_NAME_EXAMPLE
+    property :disclosure_date, type: :string, description: DISCLOSURE_DATE_EXAMPLE, example: DISCLOSURE_DATE_EXAMPLE
+    property :rank, type: :integer, description: RANK_DESC, example: RANK_EXAMPLE
+    property :type, type: :string, description: TYPE_DESC, example: TYPE_EXAMPLE, enum: TYPE_ENUM
+    property :description, type: :string, description: DESCRIPTION_DESC, example: DESCRIPTION_EXAMPLE
+    property :author, description: AUTHOR_DESC, example: AUTHOR_EXAMPLE, type: :array do items type: :string end
+    property :references, description: REFERENCE_DESC, example: REFERENCES_EXAMPLE, type: :array do items type: :string end
+    property :is_client, type: :boolean, description: IS_CLIENT_DESC, example: IS_CLIENT_EXAMPLE
+    property :is_server, type: :boolean, description: IS_SERVER_DESC, example: IS_SERVER_EXAMPLE
+    property :platform, type: :string, description: PLATFORM_DESC, example: PLATFORM_EXAMPLE
+    property :arch, type: :string, description: ARCH_DESC, example: ARCH_EXAMPLE
+    property :rport, type: :string, description: PORT_DESC, example: RPORT_EXAMPLE
+    property :mod_time, type: :string, description: MOD_TIME_DESC, example: MOD_TIME_EXAMPLE
+    property :ref_name, type: :string, description: REF_NAME_DESC, example: REF_NAME_EXAMPLE
+    property :path, type: :string, description: PATH_DESC, example: PATH_EXAMPLE
+    property :is_install_path, type: :boolean, description: IS_INSTALL_PATH_DESC, example: IS_INSTALL_PATH_EXAMPLE
+    property :targets, description: TARGET_DESC, example: TARGETS_EXAMPLE, type: :array do items type: :string end
+  end
+
+
+
+  swagger_path '/api/v1/modules' do
+    # Swagger documentation for /api/v1/modules GET
+    operation :get do
+      key :description, 'Search Metasploit modules using keyword filters.'
+      key :tags, [ 'module' ]
+
+      parameter do
```

Comment:
Is it possible to change these to in-line properties since there are so many of them?

Reply:
I cleaned this up some, and also renamed this file from `module_api_doc` to `module_search_api_doc` for consistency.